datasets/ir/business_registry/crawler.py

```diff
@@ -0,0 +1,156 @@
+from lxml import etree
+from urllib.parse import urljoin
+
+from zavod import Context, Entity, helpers as h
+from zavod.shed.zyte_api import fetch_html
+
+
+def unblock_validator(doc: etree._Element) -> bool:
+    return doc.find(".//div[@class='o-grid']") is not None
+
+
+def extract_text(doc, xpath_query):
+    result = doc.xpath(xpath_query)
+    return result[0].strip() if result else None
```

Comment:
whenever you get a list of things, be careful not to assume it will have exactly one element, or never more than one. In this case, affiliates can have more than one span. I like to just stick an `assert len(thing) == 0` in there when it would be a surprise if there are more or less elements. That sometimes helps me catch invalid assumptions.